aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -262,9 +263,12 @@ def _get_expiry_timestamp(self) -> int:
             unix timestamp of expiry date for idempotency record
 
         """
-        now = datetime.datetime.now()
-        period = datetime.timedelta(seconds=self.expires_after_seconds)
-        return int((now + period).timestamp())
+        if self.backend == "redis":
+            return self.expires_after_seconds
+        else:
+            now = datetime.datetime.now()
+            period = datetime.timedelta(seconds=self.expires_after_seconds)
+            return int((now + period).timestamp())
```

Comment:
Use [tell don't ask](https://martinfowler.com/bliki/TellDontAsk.html) here. Instead of storing the type of backend, just delegate the `get_expiry_timestamp` to the different backends, and let them deal with the details.